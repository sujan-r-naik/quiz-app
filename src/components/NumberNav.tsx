import Button from "@mui/material/Button";

const NumberNav = (props: {
  currIndex: number;
  mapIndex: number;
  setCurrentIndex: React.Dispatch<React.SetStateAction<number>>;
  qNumber: number;
  status: boolean;
}) => {
  return (
    <Button
      role="number-nav"
      sx={{
        backgroundColor: props.status
          ? "lightgreen"
          : props.currIndex === props.mapIndex
          ? "#b8d2fc"
          : "#eee",
        mx: 3,
      }}
      onClick={() => props.setCurrentIndex(props.mapIndex)}
    >
      {props.qNumber}
    </Button>
  );
};

export default NumberNav;
