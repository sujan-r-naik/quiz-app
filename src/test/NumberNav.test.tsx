import "@testing-library/jest-dom/extend-expect";
import { screen, render, fireEvent } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import NumberNav from "../components/NumberNav";
import renderer from "react-test-renderer";

test("Snapshot of NumberNav component", () => {
  const comp = renderer.create(
    <BrowserRouter>
      <NumberNav
        qNumber={4}
        setCurrentIndex={() => {}}
        status={false}
        currIndex={4}
        mapIndex={3}
      />
    </BrowserRouter>
  );
  let tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});

describe("In Quiz,", () => {
  test("does the sub component numbernav work", () => {
    render(
      <BrowserRouter>
        <NumberNav
          qNumber={4}
          setCurrentIndex={() => {}}
          status={false}
          currIndex={4}
          mapIndex={3}
        />
      </BrowserRouter>
    );

    let btn = screen.getByRole("number-nav");
    expect(btn).toBeInTheDocument();

    expect(fireEvent.click(btn)).toBeTruthy();
  });
});
