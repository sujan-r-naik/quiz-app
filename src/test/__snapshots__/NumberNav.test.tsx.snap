// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of NumberNav component 1`] = `
<button
  className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-3kyz66-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="number-nav"
  tabIndex={0}
  type="button"
>
  4
</button>
`;
